source/packages/services/certificatevendor/src/certificates/certificates.service.ts

```diff
@@ -441,5 +451,32 @@ export class CertificateService {
         return policies
 
     }
+    
+    private async getSSMCertificate(csr: string, caPem: string): Promise<string> {
+        
+        const caKey:string = await this.getCAKey(this.caCertificateId);
+        const certificate:string = await this.createCertificateFromCsr(csr, caKey, caPem);
+        return certificate;
+    }
+    
+    private async getACMCertificate(csr: string, caArn: string): Promise<string> {
```

Comment:
Missing device certificate metadata.
Need to validate response for ACM errors. from memory ACMPCA quotas are very low and are quick to throttle this needs to be accounted for.
Need to validate the full certificate chain.
See [useACMPCAProcessor.ts](https://github.com/aws/aws-connected-device-framework/blob/main/source/packages/services/provisioning/src/things/steps/useACMPCAProcessor.ts) for sample createCert method that you can use

Reply:
>Missing device certificate metadata.
Do you mean the parameters of following? 
"certInfo": {                       // optional. But mandatory if no csr was provided.
            "commonName": "?",              // optional
            "organization": "?",            // optional
            "organizationalUnit": "?",      // optional
            "locality": "?",                // optional
            "stateName": "?",               // optional
            "country": "?",                 // mandatory
            "emailAddress": "?",            // optional
            "daysExpiry": ?                 // optional
I think we do not need those parameters for CSR case because those are already included in CSR. 

>Need to validate response for ACM errors. from memory ACMPCA quotas are very low and are quick to throttle this needs to be accounted for.

Thanks. I used the function acmpca.waitFor('certificateIssued', {CertificateAuthorityArn: caArn, CertificateArn: data.CertificateArn }).promise(); instead of directly calling to getCertiifcate(). When acmpca.waitFor returns err, we just throw error. 

>Need to validate the full certificate chain.
I will change like this.
${getResponse.Certificate}\n${getResponse.CertificateChain}`;

Reply:
added change. Please review. https://github.com/aws/aws-connected-device-framework/pull/93/commits/6c21c7501abb92e72a103530cd7037d45250e144

Reply:
Based on the conversation, I have added the parameters for device certificate metadata. Please check.